build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java

```diff
@@ -147,8 +151,8 @@ public BuildSetTask build(BuildConfigurationSet buildConfigurationSet, User user
                 rebuildAll,
                 getBuildStatusChangedEventNotifier(),
                 () -> datastoreAdapter.getNextBuildRecordId());
-        BuildSetStatusChangedEvent event = buildSetTask.setStatus(BuildSetStatus.NEW);
-        buildSetStatusChangedEventNotifier.fire(event);
+        notifyBuildSetStatusChanged(buildSetTask, BuildSetStatus.NEW);
```

Comment:
Well yes I've had something like you suggested but if it contains `setStatus` then it will need to return `BuildTask`. And yes we can add overloaded version of this method to accept some String as status description as well. Don't know what's more readable...